simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -135,9 +232,11 @@ public void pushAdd(Collector collector, String job) throws IOException {
    * Pushes all metrics in a registry, replacing only previously pushed metrics of the same name, job and grouping key.
    * <p>
    * This uses the POST HTTP method.
-  */
+   */
   public void pushAdd(CollectorRegistry registry, String job, Map<String, String> groupingKey) throws IOException {
-    doRequest(registry, job, groupingKey, "POST");
+    HttpPost request = new HttpPost();
+    request.setEntity(createOutputEntity(registry));
```

Comment:
Okay but i think its not a good solution to put this in doRequest, because it depends on the request if we ne a HttpPost, HttpPut or HttpDelete.
May we could move the setEntity part to doRequest, pass the registry and check if we got a registry object in doRequest.
What do you think?